### DISCUSSION TITLE
How to mapp a attribute instantiate with dataclass __post_init__ with imperative mapping?

### DISCUSSION THREAD

**Discussion by jalvespinto** [sqlalchemy/sqlalchemy#7454] [Usage Questions]
I am trying to mapp a @dataclass that has an attribute that depends on the values of other attributes, so I placed it in a __post_init__ method, but now this attribute is not recognized by sqlalchemy? I am using imperative mapping? What should I do to make it work?

**Comment by CaselIT** [comment]
Hi,

An example of what you are doing would help understand the use case

**Comment by jalvespinto** [comment]
Hi, tks for the quick reply. Actually my mistake was on how to use __post_init__ and it is working now. tks again
